test/jdk/javax/swing/text/BoxView/bug6494356.java

```diff
@@ -51,37 +53,48 @@ public class bug6494356 {
     private static final CountDownLatch latch = new CountDownLatch(1);
 
     public static void main(final String[] args) throws Exception {
-        SwingUtilities.invokeLater(new Runnable() {
-            public void run() {
-                ep = new JEditorPane();
-                ep.setEditorKitForContentType("text/html", new MyEditorKit());
-                ep.addPropertyChangeListener("page", new PropertyChangeListener() {
-                    public void propertyChange(PropertyChangeEvent pce) {
-                        if (pce.getPropertyName().equals("page")) {
-                            latch.countDown();
+        final Path file = Path.of("bug6494356.html");
+
+        try {
+            SwingUtilities.invokeLater(new Runnable() {
+                public void run() {
+                    ep = new JEditorPane();
+                    ep.setEditorKitForContentType("text/html", new MyEditorKit());
+                    ep.addPropertyChangeListener("page", new PropertyChangeListener() {
+                        public void propertyChange(PropertyChangeEvent pce) {
+                            if (pce.getPropertyName().equals("page")) {
+                                latch.countDown();
+                            }
                         }
-                    }
-                });
-                JFrame f = new JFrame();
-                f.setTitle("6494356");
-                f.setSize(new Dimension(
+                    });
+                    JFrame f = new JFrame();
+                    f.setTitle("6494356");
+                    f.setSize(new Dimension(
                         Toolkit.getDefaultToolkit().getScreenSize().width, 600));
-                f.setContentPane(ep);
-                f.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-                f.setVisible(true);
-                File file = new File("file:" + testSrc + "/" + "bug6494356.html");
-                try {
-                    ep.setPage(file.toString());
-                } catch (Exception ex) {
-                    testPassed = false;
-                    throw new RuntimeException(ex);
+                    f.setContentPane(ep);
+                    f.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+                    f.setVisible(true);
+                    try (Writer writer = Files.newBufferedWriter(file)) {
+                        writer.write("<p>Paragraph</p>");
+                    } catch (Exception e) {
```

Comment:
I suggest creating the file *before* calling `SwingUtilities.invokeLater`.

If it throws an exception, the test fails right away, you won't even run the code `invokeLater`.